### ISSUE TITLE
LSP false positive with overloads in subclass

### ISSUE THREAD

**Issue by hauntsaninja** [microsoft/pyright#9635]
I believe the following is type safe:

```python
from typing import overload

class A: ...
class B: ...

class Parent:
    def dst(self, x: A | B) -> int | None: ...

class Child(Parent):
    @overload
    def dst(self, x: A) -> int: ...
    @overload
    def dst(self, x: B) -> None: ...
```

pyright reports:
```
Method "dst" overrides class "Parent" in an incompatible manner
  No overload signature in override is compatible with base method  (reportIncompatibleMethodOverride)
```

mypy has a similar false positive

**Comment by erictraut** [comment]
I think pyright's (and mypy's) behavior here is justified given the current typing spec and the rules spelled out in the [Callables](https://typing.readthedocs.io/en/latest/spec/callables.html#assignability-rules-for-callables) section.

If you'd like to propose an update to the typing spec, please do so in the typing forum. Short of such a change, I don't plan to make a change here in pyright.
